### PR TITLE
(CDAP-16008) Added Kerberos support to Remote Hadoop provisioner

### PR DESCRIPTION
**Pull request by chtyim**
Two new fields are added to the remote Hadoop provisioner for specifying Kerberos principal and keytab file path. The keytab file has to be presented in the SSH host where the provisioner is configured to talk to.

### CODE REVIEW

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/RemoteExecutionJobMain.java [hunk 1]
```diff
@@ -80,17 +85,32 @@ public static void main(String[] args) throws Exception {
...
    */
   private void doMain(String[] args) throws Exception {
-    if (args.length < 1) {
-      throw new IllegalArgumentException("Missing runId from the first argument");
+    Map<String, String> arguments = RuntimeArguments.fromPosixArray(args);
+    if (!arguments.containsKey(RUN_ID)) {
+      throw new IllegalArgumentException("Missing argument " + RUN_ID);
+    }
+
+    if (UserGroupInformation.isSecurityEnabled()) {
```
Comment (yaojiefeng) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/RemoteExecutionJobMain.java:93]: Curious what sets this property, is this coming from the cluster environment?

**Review by yaojiefeng — APPROVED** [review]
Just one question, LGTM

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/RemoteExecutionJobMain.java (hunk 1 above)
Comment (chtyim) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/RemoteExecutionJobMain.java:93]: Yes. It is coming from the Hadoop configuration on the cluster.

**Pull request merged**